### ISSUE TITLE
Touch/pinch rotate not working properly on some platforms since v4.0

### ISSUE THREAD

**Issue by rsimon** [openseadragon/openseadragon#2319]
I noticed that touch rotate functionality no longer works properly on some devices since the introduction of version 4.0. (Everything's fine in 3.x). Devices I tested with are Chrome on Android, and Chrome + Firefox on an Ubuntu hybrid notebook (touch screen). Things looked ok on a recent iOS iPad.

Another issue - which may or may not be related: I'm building an app which layers an SVG element on top of OSD, and keeps it in sync by listening to the `update-viewport` event and grabbing the viewport state. It looks like the rotation is no longer updated live, even when using `viewport.getRotation(true)`.

**Comment by pearcetm** [comment]
> I noticed that touch rotate functionality no longer works properly on some devices since the introduction of version 4.0.

Without knowing exactly what behavior you're seeing, I'd guess this might be related to the following code in `viewer.js`:

```
    if ( gestureSettings.pinchRotate && !canvasPinchEventArgs.preventDefaultRotateAction ) {
          // Pinch rotate
          var angle1 = Math.atan2(event.gesturePoints[0].currentPos.y - event.gesturePoints[1].currentPos.y,
              event.gesturePoints[0].currentPos.x - event.gesturePoints[1].currentPos.x);
          var angle2 = Math.atan2(event.gesturePoints[0].lastPos.y - event.gesturePoints[1].lastPos.y,
              event.gesturePoints[0].lastPos.x - event.gesturePoints[1].lastPos.x);
          this.viewport.setRotation(this.viewport.getRotation() + ((angle1 - angle2) * (180 / Math.PI)));
      }
```
In particular, I'd imagine both `getRotation` and `setRotation` should be called  with `current = true` in the setting of pinch-based rotation.

> ...listening to the `update-viewport` event and grabbing the viewport state. It looks like the rotation is no longer updated live, even when using `viewport.getRotation(true)`.

That's odd. I'm not sure why that would be happening. In a test viewer (4.0.0), in the console, these lines result in live updates to the animated rotation being printed out:
```
viewer.addHandler('update-viewport',()=>console.log(viewer.viewport.getRotation(true)));
viewer.viewport.rotateTo(30);
```

**Comment by rsimon** [comment]
The behavior I'm seeing is a bit difficult to describe for me (except as "odd"). I'm attaching a screen cap. What you're seeing is me slowly rotating with two fingers clockwise, at constant rate.

It's not really rotating and then suddenly snaps into the opposite direction. Behavior on my laptop is less pronounced but still off.

https://user-images.githubusercontent.com/470971/226726987-c2d7bfe1-7750-4a19-960e-ba3c17ba8bd6.mp4

**Comment by pearcetm** [comment]
I think what you're seeing could be consistent with the line I highlighted above. The change in rotation behavior introduced in 4.0.0 was to animate changes in rotation by default. Per the [release notes]:(https://github.com/openseadragon/openseadragon/releases/tag/v4.0.0)
> BEHAVIOR: Setting the viewport rotation now animates by default (pass false for the new immediately parameter to disable) (https://github.com/openseadragon/openseadragon/pull/2136 @jonasengelmann)

I don't have access to a touch-enabled device for development/testing, but I would guess that updating the code above to this would help:

`this.viewport.setRotation(this.viewport.getRotation(true) + ((angle1 - angle2) * (180 / Math.PI)), true);`

An added enhancement would be to rotate around the center point of the gesture:

```
centerPt = this.viewport.pointFromPixel( event.center, true );
this.viewport.rotateTo(this.viewport.getRotation(true) + ((angle1 - angle2) * (180 / Math.PI)), centerPt, true);
```

@rsimon Would you be able to test these changes out, and possibly make a pull request if they solve the problem you've identified?

**Comment by rsimon** [comment]
Thanks - this improved behavior **a lot**! I just sent a PR.

**Comment by iangilman** [comment]
Beautifully sleuthed!

**Comment by iangilman** [comment]
@pearcetm Good catch! I've now fixed it in https://github.com/openseadragon/openseadragon/releases/tag/v4.0.0 and https://github.com/openseadragon/openseadragon/blob/v4.0.0/changelog.txt. Thank you :)

**Comment by iangilman** [comment]
Fixed by #2324.
